File: lib/domain/models/tour_stop.dart
import 'tour_stop_prestation.dart';

class TourStop {
  final int id;
  final int tourId;
  final int? clientId;
  final String clientNameSnapshot;
  final int orderIndex;
  final int estimatedArrivalMinutes;
  final int estimatedDepartureMinutes;
  final List<TourStopPrestation> plannedPrestations;
  final List<TourStopPrestation>? actualPrestations;
  final String? interventionNote;
  final int feeShareCents;

  const TourStop({
    required this.id,
    required this.tourId,
    required this.clientId,
    required this.clientNameSnapshot,
    required this.orderIndex,
    required this.estimatedArrivalMinutes,
    required this.estimatedDepartureMinutes,
    required this.plannedPrestations,
    this.actualPrestations,
    this.interventionNote,
    required this.feeShareCents,
  });
}

File: src/domain/models/tour-stop.ts
import { z } from 'zod';
import { TourStopService } from './tour-stop-service';
import { Payment } from './payment';

export const TourStop = z.object({
  id: z.string(),
  tourId: z.string(),
  clientId: z.string(),
  clientNameSnapshot: z.string().nullable(),
  ordering: z.number().int(),
  arrivalMinutes: z.number().int().nullable(),
  departureMinutes: z.number().int().nullable(),
  estimatedMinutes: z.number().int().nullable(),
  feeShareCents: z.number().int().nullable(),
  plannedServices: z.array(TourStopService),
  actualServices: z.array(TourStopService).nullable(),
  notes: z.string().nullable(),
  completedAt: z.string().nullable(),
  payment: Payment,
});

export type TourStop = z.infer<typeof TourStop>;
